test/E2ETest/TestHelpers.cs

```diff
@@ -221,5 +197,106 @@ await using (cancellation.Token.Register(() => stoppedTask.TrySetCanceled()))
                 }
             }
         }
+
+        public static void CompareConfigApplications(ConfigApplication expected, ConfigApplication actual)
```

Comment:
pref: some other class named `TyeAssert` or something like that is better than putting a bunch of random stuff in *helpers*

Reply:
I see the vision, and it's definitely better.